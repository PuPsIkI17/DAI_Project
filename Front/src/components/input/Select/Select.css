.select {
    font-size: 16px;
    margin: 0;
    text-align: left;
}

.select>label {
    display: inline-block;
    font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
    font-size: 16px;
    line-height: 1.5;
    color: #333;
}

.select>select {
    font-family: "Roboto", sans-serif;
    background: #f2f2f2;
    width: 100%;
    margin: 0 0 5px;
    padding: 10px;
    box-sizing: border-box;
    font-size: 16px;
    border: 1px solid #285e8e;
    border-radius: 5px;
}