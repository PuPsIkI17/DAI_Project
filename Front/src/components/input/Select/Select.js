import React from 'react';
import PropTypes from 'prop-types';
import './Select.css'
export default class TextInput extends React.Component {
    render() {
        const options = this.props.options;
        let result = options.map(option => {
            return <option key={option} value={option}>{option.charAt(0).toUpperCase() + option.substring(1)}</option>
        })
        return (
            <div className="select">
                <label htmlFor={this.props.inputId}>{this.props.label}</label>
                <select id={this.props.id} defaultValue={this.props.defaultValue}>
                    {result}
                </select>
            </div>
        )
    }
}
TextInput.propTypes = {
    label: PropTypes.string,
    id: PropTypes.string,
    options: PropTypes.array
}
TextInput.defaultProps = {
    label: 'Label',
    id: 'defaultInput',
    options: ['Option 1'],
    defaultValue: ""
}