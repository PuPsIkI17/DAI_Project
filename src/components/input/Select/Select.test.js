import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Select from './Select';

describe('Select Component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Select />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders the UI as expected', () => {
        const select = renderer
            .create(<Select />)
            .toJSON();
        expect(select).toMatchSnapshot();
    });
})