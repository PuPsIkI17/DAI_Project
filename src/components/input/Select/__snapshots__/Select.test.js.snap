// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Component renders the UI as expected 1`] = `
<div
  className="select"
>
  <label>
    Label
  </label>
  <select
    defaultValue=""
    id="defaultInput"
  >
    <option
      value="Option 1"
    >
      Option 1
    </option>
  </select>
</div>
`;
